core/src/main/java/org/infinispan/configuration/cache/IndexingConfigurationBuilder.java

```diff
@@ -29,11 +29,12 @@
  */
 public class IndexingConfigurationBuilder extends AbstractConfigurationChildBuilder implements Builder<IndexingConfiguration>, ConfigurationBuilderInfo {
 
-   private static final String DIRECTORY_PROVIDER_SUFFIX = ".directory_provider";
+   private static final String DIRECTORY_PROVIDER_SUFFIX = "directory.type";
 
+   // TODO Check if *KEY1* can be used
    private static final String DIRECTORY_PROVIDER_KEY1 = "hibernate.search.default.directory_provider";
```

Comment:
I suspect the suffix ```directory_provider``` here also must become ```directory.type```

Reply:
Doing the change now... Anyway the full property path is `hibernate.search.backends.infinispan_backend.directory.type`, see `org.infinispan.search.mapper.mapping.impl.IndexProperties`, in particular the line:

``` java
onfigurationPropertySource basePropertySource =
            propertyChecker.wrap(ConfigurationPropertySource.fromMap(backendProperties))
                  .withPrefix(BACKEND_PROPERTIES_PREFIX);
```

Of course you can arrange the simplification of the path as you wish, I did here an example of that.